app/src/org/commcare/network/SimpleGetRequest.java

```diff
@@ -0,0 +1,88 @@
+package org.commcare.network;
+
+import org.commcare.logging.AndroidLogger;
+import org.commcare.utils.GlobalConstants;
+import org.javarosa.core.services.Logger;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Authenticator;
+import java.net.HttpURLConnection;
+import java.net.PasswordAuthentication;
+import java.net.URL;
+
+/**
+ * Makes simple redirect-following GET requests that can be authenticated or not
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+class SimpleGetRequest {
+
+    protected static InputStream makeRequest(final String username,
+                                          final String password,
+                                          URL url) throws IOException {
+        if (username == null || password == null) {
+            // clear any prior set authenticator to make unauthed requests
+            Authenticator.setDefault(null);
+        } else {
+            // make authenticated requests
+            Authenticator.setDefault(new Authenticator() {
+                @Override
+                protected PasswordAuthentication getPasswordAuthentication() {
+                    return new PasswordAuthentication(
+                            HttpRequestGenerator.buildDomainUser(username),
+                            password.toCharArray());
+                }
+            });
+        }
+        return makeModernRequest(url);
```

Comment:
A bit confused about the null vs. IOException result here, might be worth a comment explaining. So we return null if the exception is an auth issue in which case we fall back to the apache libraries, otherwise we bubble the IOException up to the top level for handling? Does the Auth issue not consistently through a more specific IOException we can catch? The combination of null and exception just feels a bit clunky to me